src/mgr/ActivePyModules.cc

```diff
@@ -985,6 +1094,161 @@ PyObject* ActivePyModules::get_perf_schema_python(
+        Formatter::ObjectSection counter_section(f, key.c_str()); // Main Object Section
+        std::optional<Formatter::ArraySection> array_section;
+
+        for (auto ctr_inst_iter : state->perf_counters.instances) {
+          /*
+              The path of the counter can either be:
+                - labeled counter path: "osd_scrub_sh_repl^@level^@shallow^@pooltype^@replicated^@.successful_scrubs_elapsed"
+                - unlabeled counter path: "osd.stat_bytes"
+              
+              For the above counters:
+                - key_names are: 'osd_scrub_sh_repl' and 'osd'
+                - counter names are: 'successful_scrubs_elapsed' and 'stat_bytes'
+
+          */
+          const auto &counter_name_with_labels = ctr_inst_iter.first;
+          auto type = state->perf_counters.types[counter_name_with_labels];
+
+          // create a vector of labels i.e [(level, shallow), (pooltype, replicated)]
+          vector<pair<string_view,string_view>> key_labels;
+          for (auto label : ceph::perf_counters::key_labels(counter_name_with_labels)) {
+            if (!label.first.empty()){
+              key_labels.push_back(label);
+            }
+          }
+
+          // Extract the key names from the counter path, these key names form
+          // the main object section for their counters
+          string key_name_without_counter;
+          if(!key_labels.empty()){
+            // key_name, osd_scrub_sh_repl
+            key_name = ceph::perf_counters::key_name(counter_name_with_labels);
+          } else {
+            size_t pos = counter_name_with_labels.find('.');
+            key_name_without_counter = counter_name_with_labels.substr(0, pos);
+            key_name = key_name_without_counter; // key_name, osd
+          }
+
+          /*
+            Construct a schema in the following format
+            {
+              "osd": [
+                {
+                  "labels": {},
+                  "counters":{
+                    "stat_byte": {
+                      "description": "",
+                      "nick": "",
+                      ...
+                    }
+                  }
+                }
+              ],
+              "osd_scrub_sh_repl":[
+                {
+                  "labels": {                         <---- 'label' section 
+                    "level": "shallow",
+                    "pooltype": "replicated"
+                  },
+                  "counters":{                        <---- 'counters' section
+                    "successful_scrubs_elapsed":{     <---- 'sub counter' section
+                      "description": "",
+                      "nick": "",
+                      ...
+                    }
+                  }
+                }                                     <---- 'counter object' close                                
+              ]
+            }
+          */
+        
+        if (prev_key_name != key_name){
```

Comment:
here and elsewhere: a blank between the closing ')' and the '{'.
I'd suggest using clang-format. I can supply the conf file.

Reply:
I would really love the conf for the clang-format! Can I please bother you with sending that my way ^^'

Reply:
not fixed?

Reply:
fixed now!